contracts/tests/test_deposit_locker.py

```diff
@@ -213,5 +166,190 @@ def test_event_slash(
         fromBlock=latest_block_number
     ).get_all_entries()[0]["args"]
 
-    assert event["validator"] == malicious_validator_address
+    assert event["slashedDepositor"] == malicious_validator_address
     assert event["slashedValue"] == deposit_amount
+
+
+@attr.s(auto_attribs=True)
+class Env:
+    deposit_locker: Any
+    slasher: Any
+    proxy: Any
+    depositors: List[Any]
+    other_accounts: List[Any]
+
+    def register_depositor(self, depositor):
+        return self.deposit_locker.functions.registerDepositor(depositor).transact(
+            {"from": self.proxy}
+        )
+
+    def register_all_depositors(self):
+        for d in self.depositors:
+            self.register_depositor(d)
+
+    def deposit(self, value_per_depositor, total_value):
+        return self.deposit_locker.functions.deposit(value_per_depositor).transact(
+            {"from": self.proxy, "value": total_value}
+        )
+
+    def slash(self, depositor):
+        return self.deposit_locker.functions.slash(depositor).transact(
+            {"from": self.slasher}
+        )
+
+    def withdraw(self, depositor):
+        return self.deposit_locker.functions.withdraw().transact({"from": depositor})
+
+    def can_withdraw(self, depositor):
+        return self.deposit_locker.functions.canWithdraw(depositor).call()
+
+
+@pytest.fixture()
+def testenv(deploy_contract, accounts, web3):
+    """return a initialized testenv instance"""
+    deposit_locker = deploy_contract("DepositLocker")
+    slasher = accounts[1]
+    proxy = accounts[2]
+    depositors = accounts[3:6]
+    other_accounts = accounts[6:]
+    deposit_locker.functions.init(web3.eth.blockNumber + 50, slasher, proxy).transact(
```

Comment:
You could also have tests showing how functions depending on values set in `init()` should fail if the contract is not initialised.

Reply:
e458e701a90b59e31ef1ff3663751056f063c1e5

Reply:
though I do not think they provide much value. One could safely remove the isInitialized modifier. If you tried to call any of the functions they would fail for different reasons.